scripts/reference_machine_run_all.sh

```diff
@@ -73,29 +73,70 @@ restore_cpu () {
   cpupower frequency-set -u "$MAX_FREQ" > /dev/null 2>&1
 }
 
-setup_disk () {
-  local file="$1"
-  local mountpoint="$2"
-  local USER_ID="$3"
-  local GROUP_ID="$4"
-  # create a simulated ext4 formatted block device in memory
-  dd if=/dev/zero of="${file?err_unset}" bs=1024 count="$((1024**2*10))" && IMG_CREATED="true"
-  LOOP_DEV="$(losetup -f)"
-  losetup --sector-size=4096 --direct-io=on "${LOOP_DEV}" "${file}" && LOOP_CREATED="true"
-  mkfs.ext4 -O ^has_journal -E nodiscard "${LOOP_DEV}"
-  mount -odefaults,noatime "${LOOP_DEV}" "${mountpoint?err_unset}" && DEVICE_MOUNTED="true"
-  chown -R "${USER_ID}:${GROUP_ID}" "${mountpoint}"
+setup_disk() {
+  # create a simulated vfat formatted brd block device in memory
+  local mountpoint="$1"
+  local USER_ID="$2"
+  local GROUP_ID="$3"
+  DEVICE="/dev/ram0"
+  # rd_size in KiB, allocate 1GiB
+  modprobe brd rd_nr=1 rd_size="$((1024**2*1))" max_part=1 && BRD_CREATED="true"
+  # fill with 1GiB of random data
+  dd if=/dev/urandom of="${DEVICE}" bs=1024 count="$((1024**2*1))"
+  # use FAT32 because it supports -o sync, logical sector size == brd physical sector size == page size == 4096
+  mkfs.vfat -F32 -S4096 "${DEVICE}"
```

Comment:
Aren't we losing some performance on disk with FAT32 ?

Reply:
The throughput and iops values in the cgroup settings have been adjusted to match the previous performance values in spite of using FAT32. The machine benchmark sequential write / random write values are within +/- 1% of previous values and now much more consistent. The point of limiting disk throughput and using RAM backed storage was to get consistent and repeatable numbers. This unfortunately did not work with EXT4 as cgroup limits only work for synchronous / O_DIRECT io. (Read more here https://andrestc.com/post/cgroups-io/)